pegen/pegen.c

```diff
@@ -304,6 +304,49 @@ endmarker_token(Parser *p)
     return expect_token(p, ENDMARKER);
 }
 
+static const char *reserved_keywords[] = {
+    "and",
+    "as",
+    "assert",
+    "async",
+    "await",
+    "break",
+    "class",
+    "continue",
+    "def",
+    "del",
+    "elif",
+    "else",
+    "except",
+    "finally",
+    "for",
+    "from",
+    "global",
+    "if",
+    "import",
+    "in",
+    "is",
+    "lambda",
+    "nonlocal",
+    "not",
+    "or",
+    "pass",
+    "raise",
+    "return",
+    "try",
+    "while",
+    "with",
+    "yield"
+};
+static int // bool
+_is_reserved_keyword(char *s) {
+    for (int i = 0; i < 32; i++) {
+        if (s[0] == reserved_keywords[i][0] && !strcmp(s, reserved_keywords[i]))
```

Comment:
Why `s[0] == reserved_keywords[i][0] `? `strcmp` is normally very optimized and has early exits. For example:

https://github.com/bminor/glibc/blob/5f72f9800b250410cad3abfeeb09469ef12b2438/string/strcmp.c#L30-L46

Reply:
It's what [Parser/parse.c](https://github.com/python/cpython/blob/558f07891170fe5173f277d3749e92d844de0a27/Parser/parser.c#L151) does. But you are right! I'll remove it.

Reply:
Now I am curious about performance benchmarks about that....maybe I am missing something. But I would advocate starting with cleaner code before we start optimizing everything :)